src/main/java/net/minestom/server/network/NetworkBufferImpl.java

```diff
@@ -555,6 +554,35 @@ static NetworkBufferImpl impl(NetworkBuffer buffer) {
         return (NetworkBufferImpl) buffer;
     }
 
+    BinaryTagWriter nbtWriter() {
+        if (this.nbtWriter == null) {
+            this.nbtWriter = new BinaryTagWriter(new DataOutputStream(new OutputStream() {
+                @Override
+                public void write(int b) {
+                    NetworkBufferImpl.this.write(BYTE, (byte) b);
+                }
+            }));
+        }
```

Comment:
Looks dirty

Reply:
Non-actionable feedback

Reply:
So I could imagine that you make a package private input and outstream or subclass